lib/common/ipc_server.c

```diff
@@ -921,6 +921,27 @@ pcmk__serve_fenced_ipc(qb_ipcs_service_t **ipcs,
     }
 }
 
+/*!
+ * \internal
+ * \brief Add an IPC server to the main loop for the pacemakerd API
+ *
+ * \param[in] cb  IPC callbacks
+ *
+ * \note This function exits with CRM_EX_OSERR if unable to create the servers.
+ */
+void
+pcmk__serve_pacemakerd_ipc(qb_ipcs_service_t **ipcs,
+                       struct qb_ipcs_service_handlers *cb)
+{
+    *ipcs = mainloop_add_ipc_server(CRM_SYSTEM_MCP, QB_IPC_NATIVE, cb);
+
+    if (*ipcs == NULL) {
+        crm_err("Couldn't start pacemakerd IPC server");
+        crm_warn("Verify pacemaker and pacemaker_remote are not both enabled.");
+        crm_exit(CRM_EX_OSERR);
```

Comment:
The other similar functions all return CRM_EX_FATAL, which makes sense to me -- it's unlikely to succeed if respawned. I don't think anything is relying on the OSERR code.

Reply:
But there is anyway nobody there to handle that restart-logic - right.
So I thought stay with where we were before for now.

Reply:
True, at this point it's better to keep the behavior -- let's just add a comment why it's different from all the other functions.

Reply:
yep - put in some explanatory comment